### PR TITLE
iCal support for events

### PR DESCRIPTION
**Pull request by acauf564**
Added ics integration for downloadable calendar events.

### CODE REVIEW

File: pages/api/schedule/ics.tsx [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+import { userIsAuthorized } from '../../../lib/authorization/check-authorization';
+import { createEvent } from 'ics';
+initializeApi();
+const db = firestore();
+
+const SCHEDULE_EVENTS = '/schedule-events';
+
+async function getIcalEvent(req: NextApiRequest, res: NextApiResponse) {
+  let event = await db.collection(SCHEDULE_EVENTS).get();
+  console.log(event);
```
Comment (Kapocsi) [inline pages/api/schedule/ics.tsx]: There are several console logs around; I recommend bumping the ones here to debug down to the [console. debug](https://developer.mozilla.org/en-US/docs/Web/API/console/debug_static) level to make it easier to filter out. Or remove them entirely if they are not needed.

File: pages/api/schedule/ics.tsx [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+  if (event.empty) {
+    console.log(event.size);
+    console.log(req.query.Event);
+  }
+  let eventName = '';
+  let icalEvents = [];
+  event.forEach((e) => {
```
Comment (Kapocsi) [inline pages/api/schedule/ics.tsx]: Nit: Avoid using the for each method and then pushing to a vector. Using [event.map](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Iterator/map) is more idiomatic.

File: pages/api/schedule/ics.tsx
Comment (Kapocsi) [inline pages/api/schedule/ics.tsx:1]: If this is an API-type route, avoid using the .tsx name because it implies the page returns some sort of react.

https://stackoverflow.com/questions/56871384/what-is-the-difference-between-ts-and-tsx-extensions-both-are-used-as-extensi

File: pages/schedule/index.tsx [hunk 3]
```diff
@@ -312,13 +314,13 @@ export default function Calendar(props: { scheduleCard: ScheduleEvent[] }) {
                   </p>
                   <p>{eventData.time}</p>
                 </div>
-                {/* <div className="">
```
Comment (Kapocsi) [inline pages/schedule/index.tsx:315]: Once again mostly a nit, 

Avoid commenting out code, git exists so we can get old code back. Having commended out code makes the diff more difficult to parse and the code harder to read

File: public/fonts/DMSans-Italic.ttf
Comment (Kapocsi) [inline public/fonts/DMSans-Italic.ttf:1]: Nit: I don't think this is to do with #1, next time let's keep this in it's own PR

**Review by Kapocsi — CHANGES_REQUESTED** [review]

File: pages/api/schedule/ics.tsx (hunk 2 above)
Comment (acauf564) [reply pages/api/schedule/ics.tsx]: I actually tried this initially but event is a QuerySnapshot (https://firebase.google.com/docs/reference/js/firestore_.querysnapshot) and that type does not have a map function. :(

Comment (acauf564) [reply pages/api/schedule/ics.tsx]: image

File: public/fonts/DMSans-Italic.ttf
Comment (acauf564) [reply public/fonts/DMSans-Italic.ttf:1]: Sorry, I didn't mean to combine these into the same PR. I realized after the fact that because I was on the same branch, it pulled this commit into this PR.

File: pages/schedule/index.tsx (hunk 3 above)
Comment (acauf564) [reply pages/schedule/index.tsx:315]: I'm not 100% sure where this came from, but I can't see it anymore on my end.

**Review by albertaspsc — APPROVED** [review]

**Review by Kapocsi — APPROVED** [review]
